client/my-sites/checkout/src/components/checkout-terms.tsx

```diff
@@ -91,54 +82,53 @@ export default function CheckoutTerms( { cart }: { cart: ResponseCart } ) {
+				<EbanxTermsOfService />
+				{ ! isGiftPurchase && <PlanTerms100Year cart={ cart } /> }
+				{ ! isGiftPurchase && <AdditionalTermsOfServiceInCart /> }
+				{ ! isGiftPurchase && <ThirdPartyPluginsTermsOfService cart={ cart } /> }
+				{ ! isGiftPurchase && <TitanTermsOfService cart={ cart } /> }
 
-					<TermsCollapsedContent>
-						{ shouldRenderFoldableCard && (
-							<FoldableCard
-								clickableHeader={ true }
-								compact
-								className="checkout__terms-foldable-card"
-								header={ translate( 'Read more' ) }
-								screenReaderText={ translate( 'Read more' ) }
-							>
-								{ shouldShowRefundPolicy && <RefundPolicies cart={ cart } /> }
-								{ shouldShowBundledDomainNotice && <BundledDomainNotice cart={ cart } /> }
-								{ shouldShowInternationalFeeNotice && <InternationalFeeNotice /> }
-								{ shouldShowJetpackSocialAdvancedPricingDisclaimer && (
-									<JetpackSocialAdvancedPricingDisclaimer />
-								) }
-							</FoldableCard>
+				<TermsCollapsedContent>
+					<CheckoutTermsReadMore>
+						{ shouldShowRefundPolicy && <RefundPolicies cart={ cart } /> }
+						{ shouldShowBundledDomainNotice && <BundledDomainNotice cart={ cart } /> }
+						{ shouldShowInternationalFeeNotice && <InternationalFeeNotice /> }
+						{ shouldShowJetpackSocialAdvancedPricingDisclaimer && (
+							<JetpackSocialAdvancedPricingDisclaimer />
 						) }
-					</TermsCollapsedContent>
-				</>
-			) : (
-				<>
-					{ ! isGiftPurchase && <DomainRegistrationAgreement cart={ cart } /> }
-					{ ! isGiftPurchase && <DomainRegistrationHsts cart={ cart } /> }
-					{ ! isGiftPurchase && <DomainRegistrationDotGay cart={ cart } /> }
-					{ shouldShowRefundPolicy && <RefundPolicies cart={ cart } /> }
-					{ shouldShowBundledDomainNotice && <BundledDomainNotice cart={ cart } /> }
-					{ ! isGiftPurchase && <TitanTermsOfService cart={ cart } /> }
-					{ ! isGiftPurchase && <ThirdPartyPluginsTermsOfService cart={ cart } /> }
-					{ ! isGiftPurchase && <PlanTerms100Year cart={ cart } /> }
-					<EbanxTermsOfService />
-					{ shouldShowInternationalFeeNotice && <InternationalFeeNotice /> }
-					{ ! isGiftPurchase && <AdditionalTermsOfServiceInCart /> }
-					{ shouldShowJetpackSocialAdvancedPricingDisclaimer && (
-						<JetpackSocialAdvancedPricingDisclaimer />
-					) }
-				</>
-			) }
+					</CheckoutTermsReadMore>
+				</TermsCollapsedContent>
+			</>
 		</Fragment>
 	);
 }
+
+/**
+ * Render a FoldableCard to contain TOS items or nothing if there are no items.
+ */
+function CheckoutTermsReadMore( { children }: { children: ReactNode } ) {
+	const translate = useTranslate();
+	// Note that this technique for finding children does not work for strings or
+	// empty fragments. Hopefully all children passed to this component are
+	// components themselves or null.
+	if ( ! children || Children.toArray( children ).filter( isValidElement ).length === 0 ) {
+		return null;
+	}
+	return (
+		<TermsCollapsedContent>
```

Comment:
Is this being rendered twice? See line 95. Not sure it really matters

Reply:
Huh, you're right.